tests/store/tracking/test_sqlalchemy_store.py

```diff
@@ -3875,3 +3876,85 @@ def test_start_trace_with_invalid_experiment_id(store: SqlAlchemyStore):
             request_metadata={},
             tags={},
         )
+
+
+def _create_trace(
+    store: SqlAlchemyStore,
+    request_id: str,
+    experiment_id=0,
+    timestamp_ms=0,
+    execution_time_ms=0,
+    status=TraceStatus.OK,
+    request_metadata=None,
+    tags=None,
+) -> TraceInfo:
+    """Helper function to create a test trace in the database."""
+    if not store.get_experiment(experiment_id):
+        store.create_experiment(store, experiment_id)
+
+    with mock.patch.object(store, "_generate_trace_request_id", lambda: request_id):
+        trace_info = store.start_trace(
+            experiment_id=experiment_id,
+            timestamp_ms=timestamp_ms,
+            request_metadata=request_metadata or {},
+            tags=tags or {},
+        )
+
+    store.end_trace(
+        request_id=request_id,
+        timestamp_ms=timestamp_ms + execution_time_ms,
+        status=status,
+        request_metadata={},
+        tags={},
+    )
+    return trace_info
+
+
+def test_set_and_delete_tags(store: SqlAlchemyStore):
+    exp1 = store.create_experiment("exp1")
+    request_id = "tr-123"
+    _create_trace(store, request_id, experiment_id=exp1)
+
+    assert store.get_trace_info(request_id).tags == {}
+
+    store.set_trace_tag(request_id, "tag1", "apple")
```

Comment:
Can we add a mutation validation here (another condition where we're updating tag1's value and validating the update occurred prior to deleting it?